lib/side_header_list_view.dart

```diff
@@ -21,97 +19,89 @@ class SideHeaderListView extends StatefulWidget {
   final IndexedWidgetBuilder itemBuilder;
   final EdgeInsets padding;
   final HasSameHeader hasSameHeader;
-  final itemExtend;
+  final double itemExtent;
 
   SideHeaderListView({
     Key key,
     this.itemCount,
-    @required this.itemExtend,
+    @required this.itemExtent,
     @required this.headerBuilder,
     @required this.itemBuilder,
     @required this.hasSameHeader,
     this.padding,
-  })
-      : super(key: key);
+  }) : super(key: key);
 
   @override
-  _SideHeaderListViewState createState() => new _SideHeaderListViewState();
+  _SideHeaderListViewState createState() => _SideHeaderListViewState();
 }
 
 class _SideHeaderListViewState extends State<SideHeaderListView> {
   int currentPosition = 0;
+  final _controller = ScrollController();
 
   @override
-  Widget build(BuildContext context) {
-    return new Stack(
-      children: <Widget>[
-        new Positioned(
-          child: new Opacity(
-            opacity: _shouldShowHeader(currentPosition) ? 0.0 : 1.0,
-            child: widget.headerBuilder(context, currentPosition >= 0 ? currentPosition : 0),
-          ),
-          top: 0.0 + (widget.padding?.top ?? 0),
-          left: 0.0 + (widget.padding?.left ?? 0),
-        ),
-        new ListView.builder(
-            padding: widget.padding,
-            itemCount: widget.itemCount,
-            itemExtent: widget.itemExtend,
-            controller: _getScrollController(),
-            itemBuilder: (BuildContext context, int index) {
-              return new Row(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  new FittedBox(
-                    child: new Opacity(
-                      opacity: _shouldShowHeader(index) ? 1.0 : 0.0,
-                      child: widget.headerBuilder(context, index),
-                    ),
-                  ),
-                  new Expanded(child: widget.itemBuilder(context, index))
-                ],
-              );
-            }),
-      ],
-    );
+  void initState() {
+    super.initState();
+    // Reposition side view each time the main list view is scrolled
+    _controller.addListener(_reposition);
+    // In case the initial offset is not 0, we reposition the side view to the
+    // correct offset after first build
+    // TODO: this produces a flickering effect -- we should try to position it
+    // correctly before the first build
```

Comment:
Can't we do this in the build method? We might need to set some property if the position is set or not.

I'm also think if we need to do a reposition in didChangeDependencies

Reply:
So the problem is that if you try to do this anywhere before the first build (for example in `initState` or `didChangeDependencies`) it complains with `ScrollController not attached to any scroll view.`

The only workaround I could find was to wait until the build is done, so the controller is attached to a scroll view, and then update the offset. There might a better solution but not sure how exactly it can be done in `build` or `didChangeDependencies`